### PR TITLE
Vendor gardener/gardener@v1.91.1, Upgrade k8s.io/* dependencies and controller-runtime

### PR DESCRIPTION
**Pull request by hebelsan**
**How to categorize this PR?**

/area open-source
/kind enhancement
/platform azure

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other dependency
The following golang dependencies have been upgraded :
- `gardener/gardener`: `v1.87.0`->`v1.91.1`
- k8s.io/* : v0.28.3 -> v0.29.3
- sigs.k8s.io/controller-runtime: v0.16.3-> v0.17.2
- sigs.k8s.io/controller-tools v0.13.0-> v0.14.0 
```

### CODE REVIEW

File: pkg/controller/infrastructure/terraform_reconciler_test.go [hunk 1]
```diff
@@ -332,8 +332,8 @@ var _ = Describe("Actuator", func() {
 		It("should restore infrastructure with countFault and countUpdate domain values found in ProviderStatus if availability set is required", func() {
 			providerStatus.AvailabilitySets = append(providerStatus.AvailabilitySets, apiv1alpha1.AvailabilitySet{
 				Purpose:            apiv1alpha1.PurposeNodes,
-				CountFaultDomains:  pointer.Int32(6),
-				CountUpdateDomains: pointer.Int32(7),
+				CountFaultDomains:  ptr.To(int32(6)),
```
Comment (shafeeqes) [inline pkg/controller/infrastructure/terraform_reconciler_test.go:335]: You could use ptr.To[int32]()

**Comment by hebelsan** [comment]
Updated golang image master, [53514cc1](https://github.com/gardener/gardener-extension-provider-azure/commit/53514cc1)

**Comment by shafeeqes** [comment]
Can you also add `KnownCodes` to the infra reconciler, similar to https://github.com/gardener/gardener-extension-provider-aws/pull/901/commits/76d16105e4383f974b2cf02cf50c54129a6ef7f0?

**Review by kon-angelo — APPROVED** [review]
/lgtm

**Pull request merged**